### ISSUE TITLE
Stacked walls will stack their flooding and water flow rates when breached

### ISSUE THREAD

**Issue by WJohn** [FakeFishGames/Barotrauma#13469]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

**The Problem:**
When multiple walls are overlapping, damaging them will have them individually add flow rate to the room despite covering the same surface area of the hull. This can result in seemingly decorative stacked walls and hulls both contributing to flooding and causing much more extreme flooding when damaged.

In the following video (on 1.2.12.0 Unstable), the left wall would ordinarily flood at a flow rate of 543.7. The 1000 flow pump can handle this. The right wall has 5 walls stacked on top of each other, and will subsequently have a flow rate of 2718.5 despite being the exact same size of a breach.

https://github.com/FakeFishGames/Barotrauma/assets/5988156/248a6f1d-6291-49c4-81b8-1a26da0e1669

**Expected Result:**
Breach flow rate should first be determined by the surface area of the hull block being affected by a breach, and secondly by the single most damaged wall affecting that section.

**Notes:**
- This effect is even more extreme on the current live build v1.2.8.0 Winter Update hotfix 2, as breaches of this size produce flow rates of 966.66, so the right wall would produce a flow of 4833.3.
- Several submarines have decorative shell pieces overlaid on top of indoor walls, such as the back of the Humpback and Orca 2, and the command room of the Typhon 1. These rooms flood incredibly fast when those pieces are damaged.

### Reproduction steps

1. Enter the Submarine Editor
2. Load the Orca 2
3. Destroy the frontmost walls in front of the captain's Navigation Terminal
4. Observe the flow rate
5. Destroy the furthest back walls in the Engine Room
6. Compare the flow rate with the breach in the Command Room

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.2.12.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Fixed in https://github.com/FakeFishGames/Barotrauma-development/commit/f75dd4ecabe5449f66b060ade96741a057744975

**Comment by itchyOwl** [comment]
Found out it's not that easy to repro the issue without setting up special test conditions. Couldn't see any noticeable difference before and after the fix when damaging walls normally. There seemed to be some difference between regular single walls and multiple overlapping walls, but the differences were relatively subtle. It seems that the differences become bigger when the gaps are bigger (as one would expect when the issue is multiplicative).

So, I set up a test case, similar to shown in the video: 1 wall vs 5 overlapping walls in a hull with a pump with max flow of 1000.

1 wall (before or after the fix):
![image](https://github.com/user-attachments/assets/ef7089a3-a81a-4ee6-95fa-82ef5ba0c18e)
5 walls (before the fix):
![image](https://github.com/user-attachments/assets/cd497f86-2b8e-4d76-bde1-e6cb6eeac34e)
5 walls (after the fix):
![image](https://github.com/user-attachments/assets/94b8cbc9-3350-41d7-8cef-9f1ea5643539)

The results, however, are a bit different if you cut multiple smaller breaches all over the wall. Here I've tried to find the point where the water level starts rising.

1 wall before/after the fix:
![image](https://github.com/user-attachments/assets/817baf60-ad2b-4d80-a8d3-b136869c19ab)

5 walls before the fix:
![image](https://github.com/user-attachments/assets/ca5c08dd-4a7c-4b07-b807-20efcaa8deb0)

5 walls after the fix:
![image](https://github.com/user-attachments/assets/a19b9384-8a53-4288-980c-821c0573a58b)

To me the results of the latter test seems similar before and after the fix: overalapping walls _seem_ to behave a bit differently than single walls. But I wasn't entirely sure if the differences were just about the damage layer being more visible on single walls. So I re-tested it:

1 wall before/after the fix:
![image](https://github.com/user-attachments/assets/82891b1e-9e75-4e73-8c9e-70d1ecf7fdba)

5 walls before the fix:
![image](https://github.com/user-attachments/assets/1c95c0d9-b979-4acb-82f4-3b8fa7e7d6e4)
-> Seems to actually require less damage to the wall to cause the same amount of flooding (as expected?).

5 walls after the fix (first run):
![image](https://github.com/user-attachments/assets/6aea9920-af06-4c34-a6ea-f2785edf5903)
-> Seems that after the fix the required damage to the wall is the same regardless how many layers there overlapping (as expected?). The damage layer seems to be hidden behind a wall there. So I did one more test.

5 walls after the fix (second run):
![image](https://github.com/user-attachments/assets/d7b6b627-c00e-4a07-b310-76f5481c46a8)
-> Functionally the results are the same again, but now the damage layers look really weird.

So in short: I think the fix works, but there's something goofy happening with the damage layers. Not worth addressing imo.
